packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -880,13 +895,27 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
 
         unchecked {
             asset.validatorKro += assets;
+            vault.lastDepositedAt = uint128(block.timestamp);
         }
 
         if (updateTree) {
             VALIDATOR_MANAGER.updateValidatorTree(validator, false);
         }
     }
 
+    function _withdraw(address validator, uint128 assets) internal {
+        Vault storage vault = _vaults[validator];
+        if (assets > vault.asset.validatorKro) revert InsufficientAsset();
+
+        ASSET_TOKEN.safeTransfer(vault.withdrawAccount, assets);
```

Comment:
As done in `undelegate`, why don't we move this and L916 out of this internal function?